### ISSUE TITLE
Resize event in week/day view doesn't change the event data

### ISSUE THREAD

**Issue by rodrigoaguilera** [owncloud/calendar#910]
### Steps to reproduce

also reproducible on https://demo.owncloud.org/index.php/apps/calendar/

1. download and enable calendar app in owncloud
2. Create an event
3. Resize the duration of the event by dragging the bottom or top of the event
4. Reload the page

### Expected behaviour
You should see the event with the duration you set by dragging

### Actual behaviour
I can see the changes visually but if I reload the page I can see that the event is back to the state it was before.

### Server configuration
**Operating system**: debian 8

**Web server:** nginx 1.6.2

**Database:** mysql 5.6

**PHP version:** 5.6

**ownCloud version:** 8.1.1

** owncloud calendar version:** 0.7.3

**Updated from an older ownCloud or fresh install:** fresh

**List of activated apps:**
activity
pictures
notes
calendar
contacts
tasks

**The content of config/config.php:**
$CONFIG = array (
  'instanceid' => 'xxxxxxxx',
  'passwordsalt' => 'xxxxxxxxxx',
  'secret' => 'xxxxxxxxxxxx',
  'datadirectory' => '/var/www/ownclouddata',
  'overwrite.cli.url' => 'https://cl.xxxxxxxxxxxxx.com',
  'dbtype' => 'mysql',
  'version' => '8.1.1.3',
  'dbname' => 'owncloud',
  'dbhost' => 'localhost',
  'dbtableprefix' => 'oc_',
  'dbuser' => 'oc_admin',
  'dbpassword' => 'xxxxxxxxx',
  'installed' => true,
  'forcessl' => true,
  'forceSSLforSubdomains' => true,
  'mail_smtpmode' => 'smtp',
  'loglevel' => 1,
  'mail_smtpsecure' => 'ssl',
  'mail_from_address' => 'xxxxxxx',
  'mail_domain' => 'xxxxxxx.com',
  'mail_smtpauthtype' => 'LOGIN',
  'mail_smtpauth' => 1,
  'mail_smtphost' => 'xxxxxx.yyyyyyy.com',
  'mail_smtpport' => '465',
  'mail_smtpname' => 'xxxx@yyyyyyy.com',
  'mail_smtppassword' => 'xxxxxxxxxxx',
  'theme' => '',
  'maintenance' => false,
);

**Are you using external storage, if yes which one:** local/smb/sftp/...
No
**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** 
NO

### Client configuration
**Browser:**
Tested on chrome and firefox

**Operating system:**
Debian

#### Web server error log

2015/09/02 04:58:24 [crit] 28435#0: *41099 SSL_do_handshake() failed (SSL: error:14076102:SSL routines:SSL23_GET_CLIENT_HELLO:unsupported protocol) while SSL handshaking, client: 50.7.207.42, server: 0.0.0.0:443
2015/09/02 04:58:58 [crit] 28435#0: *41118 SSL_do_handshake() failed (SSL: error:14076102:SSL routines:SSL23_GET_CLIENT_HELLO:unsupported protocol) while SSL handshaking, client: 50.7.207.42, server: 0.0.0.0:443
2015/09/02 04:59:00 [crit] 28435#0: *41124 SSL_do_handshake() failed (SSL: error:14076102:SSL routines:SSL23_GET_CLIENT_HELLO:unsupported protocol) while SSL handshaking, client: 50.7.207.42, server: 0.0.0.0:443
2015/09/02 12:12:44 [crit] 28435#0: *41166 SSL_do_handshake() failed (SSL: error:14076102:SSL routines:SSL23_GET_CLIENT_HELLO:unsupported protocol) while SSL handshaking, client: 184.105.139.67, server: 0.0.0.0:443
2015/09/02 15:50:39 [error] 28435#0: *41332 access forbidden by rule, client: x.x.x.x, server: xxx.com, request: "GET /data/htaccesstest.txt HTTP/1.1", host: "xxx.com"

#### ownCloud log (data/owncloud.log)
Error	PHP	Array to string conversion at /var/www/owncloud/lib/private/template/functions.php#36	2015-09-02T13:50:36+00:00
Error	PHP	Cannot modify header information - headers already sent by (output started at /var/www/owncloud/3rdparty/sabre/http/lib/Sapi.php:70) at /var/www/owncloud/3rdparty/sabre/http/lib/Sapi.php#58	2015-09-02T10:56:38+00:00
Error	PHP	Cannot modify header information - headers already sent by (output started at /var/www/owncloud/3rdparty/sabre/http/lib/Sapi.php:70) at /var/www/owncloud/3rdparty/sabre/http/lib/Sapi.php#58	2015-09-02T10:49:58+00:00
Error	PHP	Cannot modify header information - headers already sent by (output started at /var/www/owncloud/3rdparty/sabre/http/lib/Sapi.php:70) at /var/www/owncloud/3rdparty/sabre/http/lib/Sapi.php#58	2015-09-01T17:46:29+00:00
Error	PHP	Cannot modify header information - headers already sent by (output started at /var/www/owncloud/3rdparty/sabre/http/lib/Sapi.php:70) at /var/www/owncloud/3rdparty/sabre/http/lib/Sapi.php#58	2015-09-01T10:25:07+00:00
Error	PHP	Cannot modify header information - headers already sent by (output started at /var/www/owncloud/3rdparty/sabre/http/lib/Sapi.php:70) at /var/www/owncloud/3rdparty/sabre/http/lib/Sapi.php#58	2015-08-31T19:08:54+00:00
Error	PHP	Cannot modify header information - headers already sent by (output started at /var/www/owncloud/3rdparty/sabre/http/lib/Sapi.php:70) at /var/www/owncloud/3rdparty/sabre/http/lib/Sapi.php#58

#### Browser log

console log 
Post to https://xxxx.com/index.php/apps/calendar/ajax/event/resize.php
id=1770&dayDelta=0&minuteDelta=60&lastmodified=1441042737
response:
{"lastmodified":1441042737,"status":"success"}
Event resized successfully

**Comment by georgehrke** [comment]
please use the [issue template](https://raw.githubusercontent.com/owncloud/core/master/issue_template.md <https://raw.githubusercontent.com/owncloud/core/master/issue_template.md>) as described in our [contributing guidelines](https://github.com/owncloud/core/blob/master/CONTRIBUTING.md <https://github.com/owncloud/core/blob/master/CONTRIBUTING.md>) …

**Comment by dewap** [comment]
Same problem here, Calendar 0.7.3 and ownCloud 8.1.1 (no fresh install)
Archlinux up to date / nginx 1.8.0 / php 5.6.12 / sqlite 3.8.11.1

**Comment by georgehrke** [comment]
@dewap please use the issue template ..........

**Comment by rodrigoaguilera** [comment]
updated

**Comment by georgehrke** [comment]
please provide the web server's error log. There is a reason we asking for it ...

**Comment by rodrigoaguilera** [comment]
The error that happens when I resize the event is 

Error	PHP	Array to string conversion at /var/www/owncloud/lib/private/template/functions.php#36	2015-09-02T13:50:36+00:00

**Comment by georgehrke** [comment]
and what's about the owncloud.log?

**Comment by rodrigoaguilera** [comment]
The owncloud.log (json or something )has the same entries as what I posted (the log that is in the admin interface)

**Comment by jeffrobo** [comment]
I ran across this one, too. Fortunately, it was before I updated my customers. They are constantly updating events all day long as the field techs get stuck on one job or finish a job early. They would have chewed my head off if I stuck them with this one.

Take a look in ajax/event/resize.php and go to line 31. It looks like someone was in the middle of changing it and got distracted. The line reads:
```
$dtend->getDateTime()->add($delta);
```
Using the previous version as an example, I believe the line should read:
```
$dtend->setDateTime($dtend->getDateTime()->add($delta));
```
That seems to do the trick. I can drag the bottom edge of the event to lengthen or short the time and the new time is properly updated.
